programs/bpf_loader/src/syscalls.rs

```diff
@@ -2118,976 +2285,3475 @@ fn call<'a>(
-        enable_instruction_meter: true,
-        enable_instruction_tracing: false,
-    };
+struct SyscallBigNumFromU32<'a> {
+    cost: u64,
+    compute_meter: Rc<RefCell<dyn ComputeMeter>>,
+    loader_id: &'a Pubkey,
+}
+impl<'a> SyscallObject<BpfError> for SyscallBigNumFromU32<'a> {
+    fn call(
+        &mut self,
+        bn_out_addr: u64,
+        u64_out_len: u64,
+        u32_val: u64,
+        _arg4: u64,
+        _arg5: u64,
+        memory_mapping: &MemoryMapping,
+        result: &mut Result<u64, EbpfError<BpfError>>,
+    ) {
+        question_mark!(
+            self.compute_meter
+                .consume(self.cost + calc_bignum_cost!(u64_out_len as f64)),
+            result
+        );
+        let byte_slice = question_mark!(
+            translate_slice_mut::<u8>(
+                memory_mapping,
+                bn_out_addr,
+                u64_out_len,
+                self.loader_id,
+                true
+            ),
+            result
+        );
+        let bn_bytes = BigNum::from_u32(u32_val as u32).unwrap().to_vec();
+        if bn_bytes.len() as u64 > u64_out_len {
+            *result = Err(SyscallError::BigNumberToBytesError.into())
+        } else {
+            (*byte_slice).copy_from_slice(&bn_bytes);
+            *result = Ok(0)
+        }
+    }
+}
+/// Converts a decimal string to a bignum and returns
+/// the vector bytes and if negative or not
+struct SyscallBigNumFromDecStr<'a> {
+    cost: u64,
+    compute_meter: Rc<RefCell<dyn ComputeMeter>>,
+    loader_id: &'a Pubkey,
+}
+impl<'a> SyscallObject<BpfError> for SyscallBigNumFromDecStr<'a> {
+    fn call(
+        &mut self,
+        in_dec_str_addr: u64,
+        in_size: u64,
+        out_array_addr: u64,
+        out_size_addr: u64,
+        out_negative_addr: u64,
+        memory_mapping: &MemoryMapping,
+        result: &mut Result<u64, EbpfError<BpfError>>,
+    ) {
+        question_mark!(
+            self.compute_meter
+                .consume(self.cost + calc_bignum_cost!(in_size as f64)),
+            result
+        );
 
-    macro_rules! assert_access_violation {
-        ($result:expr, $va:expr, $len:expr) => {
-            match $result {
-                Err(EbpfError::AccessViolation(_, _, va, len, _)) if $va == va && len == len => (),
-                _ => panic!(),
-            }
+        // Get the string and convert to BigNum
+        let in_dec_raw = question_mark!(
+            translate_slice::<u8>(
+                memory_mapping,
+                in_dec_str_addr,
+                in_size,
+                self.loader_id,
+                true
+            ),
+            result
+        );
+        let i = match in_dec_raw.iter().position(|byte| *byte == 0) {
+            Some(i) => i,
+            None => in_size as usize,
         };
+        let big_number = BigNum::from_dec_str(from_utf8(&in_dec_raw[..i]).unwrap()).unwrap();
+        let big_number_bytes = big_number.as_ref().to_vec();
```

Comment:
`to_vec()` here does an alloc and copy when all we need is a slice

Reply:
There is not `to_slice()` to be gotten!